python/pyarrow/pandas_compat.py

```diff
@@ -730,15 +733,18 @@ def _reconstruct_block(item, columns=None, extension_columns=None):
             block_arr, categories=item['dictionary'],
             ordered=item['ordered'])
         block = _int.make_block(cat, placement=placement,
-                                klass=_int.CategoricalBlock)
+                                klass=_int.CategoricalBlock, ndim=ndim)
     elif 'timezone' in item:
         dtype = make_datetimetz(item['timezone'])
+        cls = dtype.construct_array_type()
+        block_arr = cls._simple_new(block_arr, dtype=dtype)
```

Comment:
this line seems to cause the failures on latest pandas. So if the previous way was working, I would keep it that way (in addition, although pyarrow is already using private APIs here, `_simple_new` is even more a private implementation detail, so I would prefer not using it)

Reply:
In addition, the `dtype.construct_array_type()` line is failing on older pandas releases. So reverting this change might solve that as well.

Reply:
IIRC the motivation here was to stop passing `dtype` to make_block, since that kwarg isnt used within pandas and AFAICT this usage is the blocker to deprecating/removing it.  could be considered separate from the matter at hand, but while im in the neighborhood

Reply:
OK, I understand. 
Now, next problem: also `DatetimeArray` is not yet available on the older pandas versions ..

Reply:
(now we could do something else depending on the pandas version)